src/main/java/de/tum/in/www1/artemis/service/DataExportService.java

```diff
@@ -0,0 +1,567 @@
+        }
+    }
+
+    private void createExportForFileUploadExercises(Set<FileUploadExercise> fileUploadExercises, Path courseWorkingDir) throws IOException {
+        for (var fileUploadExercise : fileUploadExercises) {
+            createNonProgrammingExerciseExport(fileUploadExercise, courseWorkingDir);
+        }
+    }
+
+    private void addGeneralUserInformation(User user) throws IOException {
+        String[] headers = new String[] { "login", "name", "email", "registration number" };
+        CSVFormat csvFormat = CSVFormat.DEFAULT.builder().setHeader(headers).build();
+
+        try (final CSVPrinter printer = new CSVPrinter(Files.newBufferedWriter(workingDirectory.resolve("general_user_information" + CSV_FILE_EXTENSION)), csvFormat)) {
+            printer.printRecord(user.getLogin(), user.getName(), user.getEmail(), user.getRegistrationNumber());
+            printer.flush();
+
+        }
+    }
+
+    private void createProgrammingExerciseExport(ProgrammingExercise programmingExercise, Path courseDir) throws IOException {
+        Path exerciseDir = courseDir.resolve(programmingExercise.getSanitizedExerciseTitle());
+        if (!Files.exists(exerciseDir)) {
+            Files.createDirectory(exerciseDir);
+        }
+        createSubmissionsResultsExport(programmingExercise, exerciseDir);
+        RepositoryExportOptionsDTO repositoryExportOptions = new RepositoryExportOptionsDTO();
+        repositoryExportOptions.setExportAllParticipants(false);
+        repositoryExportOptions.setAnonymizeStudentCommits(false);
+        repositoryExportOptions.setFilterLateSubmissions(false);
+        repositoryExportOptions.setCombineStudentCommits(false);
+        repositoryExportOptions.setFilterLateSubmissionsIndividualDueDate(false);
+        repositoryExportOptions.setExcludePracticeSubmissions(false);
+        repositoryExportOptions.setHideStudentNameInZippedFolder(false);
+        repositoryExportOptions.setNormalizeCodeStyle(true);
+        var listOfProgrammingExerciseParticipations = programmingExercise.getStudentParticipations().stream()
+                .filter(studentParticipation -> studentParticipation instanceof ProgrammingExerciseStudentParticipation)
+                .map(studentParticipation -> (ProgrammingExerciseStudentParticipation) studentParticipation).toList();
+        List<String> exportRepoErrors = new ArrayList<>();
+        var tempRepoWorkingDir = fileService.getUniquePath(repoClonePath.toString());
+        programmingExerciseExportService.exportStudentRepositories(programmingExercise, listOfProgrammingExerciseParticipations, repositoryExportOptions, tempRepoWorkingDir,
+                exerciseDir, exportRepoErrors);
+        // we use this directory only to clone the repository and don't do this in our current directory because the current directory is part of the final data export
+        // --> we can delete it after use
+        fileService.scheduleForDirectoryDeletion(tempRepoWorkingDir, 5);
```

Comment:
Are you sure that this will always be enough time?
Can't we also explicitly delete the folder after the export is done?